chall09/estina.c

```diff
@@ -0,0 +1,96 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   estina.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: estina <estina@student.42madrid.com>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/05/09 19:45:09 by estina            #+#    #+#             */
+/*   Updated: 2020/05/10 15:34:40 by estina           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ctype.h>
+
+int		check_input(char *seconds)
+{
+	int		i;
+
+	if (!*seconds)
+		return (0);
+	i = strlen(seconds) - 1;
+	while (isspace(seconds[i]))
+		i--;
+	while (i >= 0)
+	{
+		if (!isdigit(seconds[i]))
+			return (0);
+		i--;
+	}
+	return (1);
+}
+
+unsigned long long	transform_time(void *y, int z)
+{
+	unsigned long long	x;
+
+	x = *((unsigned long long *)y) / z;
+	*((unsigned long long *)y) %= z;
+	return (x);
+}
+
+void	append_time(unsigned long long time, char *one, char *more, char *n, char **str)
+{
+	char	buf[13];
+
+	if (time)
+	{
+		sprintf(buf, "%llu", time);
+		*str = strcat(*str, buf);
+		*str = strcat(*str, time == 1 ? one : more);
```

Comment:
You could have used `strncat()` or `asprintf()` and you probably wouldn't have had buffer problems (really powerful function).

Reply:
Should check that functions, thanks!